File: JAVA/java基础学习1/小练习/回文数/src/huiwen.java
/**
 * @Auther :sofency
 * @Date: ${DATA}
 * @Description: PACKAGE_NAME
 * @Version: 1.0
 */
import javax.swing.plaf.synth.SynthInternalFrameUI;
import java.util.Scanner;
public class huiwen {
    public static void main(String[] args) {
        Scanner input=new Scanner(System.in);
        System.out.print("Enter an integer:");
        int number=input.nextInt();
        if(judge(number)){
            System.out.println(number+" is a palindrome");
        } else {
            System.out.println(number+" is a palindrome");
        }
    }
    public static boolean judge(int num){
        int[] arr=new int[100];
        int cnt=0;
        while(num!=0) {
            arr[cnt++]=num%10;
            num=num/10;
        }
        for(int i=0;i<cnt/2;i++) {
            if(arr[i]!=arr[cnt-i-1]) {
                return false;
            }
        }
        return true;
    }
}

File: Python/create/huiwen.py
#回文字符串的比较
def huiwen_str(string):
    size=len(string)
    for i in range(size//2):
        if string[i]!=string[size-1-i]:
            return False
    return True
num=input("请输入要判断的字符串或者数字:"+"\n")#输入的数字会被转化为字符串
if huiwen_str(num):
    print("回文字符串")
else:
    print("不是回文字符串")